web/src/app/services/IntegrationKeyDeleteDialog.js

```diff
@@ -69,48 +82,33 @@ export default class IntegrationKeyDeleteDialog extends React.PureComponent {
               service: {
                 ...service,
                 integrationKeys: (service.integrationKeys || []).filter(
-                  (key) => key.id !== this.props.integrationKeyID,
+                  (key) => key.id !== props.integrationKeyID,
                 ),
               },
             },
           })
         }}
       >
-        {(commit, status) => this.renderDialog(data, commit, status)}
+        {(commit, status) => renderDialog(data, commit, status)}
       </Mutation>
     )
   }
 
-  renderDialog(data, commit, mutStatus) {
-    const { loading, error } = mutStatus
-
+  function renderQuery() {
     return (
-      <FormDialog
-        title='Are you sure?'
-        confirm
-        subTitle={`This will delete the integration key: ${data.name}`}
-        caption='This will prevent the creation of new alerts using this integration key. If you wish to re-enable, a NEW integration key must be created and may require additional reconfiguration of the alert source.'
-        loading={loading}
-        errors={nonFieldErrors(error)}
-        onClose={this.props.onClose}
-        onSubmit={() => {
-          const input = [
-            {
-              type: 'integrationKey',
-              id: this.props.integrationKeyID,
-            },
-          ]
-          return commit({
-            variables: {
-              input,
-            },
-          })
-        }}
+      <Query
+        noPoll
+        query={query}
+        variables={{ id: props.integrationKeyID }}
+        render={({ data }) => renderMutation(data.integrationKey)}
```

Comment:
Since we're switching to hooks, let's replace this with `useQuery` same with `Mutation` -> `useMutation`